Plugin/Catalog/Model/Sortby.php

```diff
@@ -0,0 +1,107 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+
+namespace Nosto\Tagging\Plugin\Catalog\Model;
+
+use Magento\Catalog\Model\Category\Attribute\Source\Sortby as MagentoSortby;
+use Nosto\Tagging\Plugin\Catalog\Model\Config as NostoConfig;
+use Magento\Catalog\Model\Category as MagentoCategory;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Magento\Backend\Block\Template\Context;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\App\Request\Http;
+
+class Sortby extends Template
+{
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var NostoHelperAccount */
+    private $nostoHelperAccount;
+
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var Http $request */
+    private $request;
+
+    /**
+     * Sortby constructor.
+     * @param NostoHelperData $nostoHelperData
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param Context $context
+     * @param StoreManagerInterface $storeManager
+     * @param array $data
+     */
+    public function __construct(
+        NostoHelperData $nostoHelperData,
+        NostoHelperAccount $nostoHelperAccount,
+        Context $context,
+        StoreManagerInterface $storeManager,
+        Http $request,
+        array $data = []
+    ) {
+        $this->nostoHelperData = $nostoHelperData;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->storeManager = $storeManager;
+        $this->request = $request;
+        parent::__construct($context, $data);
+    }
+
+    public function afterGetAllOptions(MagentoSortby $sortBy, $options)
```

Comment:
Missin docblock, also `$sortBy` seems not to be used

Reply:
The first param for the method is the Interceptor, the second one are the options needed.